File: netlogo-gui/src/main/prim/_constlist.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.api.Dump;
import org.nlogo.core.LogoList;
import org.nlogo.core.Syntax;
import org.nlogo.core.Pure;
import org.nlogo.nvm.Reporter;

public final strictfp class _constlist
    extends Reporter implements Pure {
  final LogoList value;

  public _constlist(LogoList value) {
    this.value = value;
  }



  @Override
  public LogoList report(final org.nlogo.nvm.Context context) {
    return value;
  }

  @Override
  public String toString() {
    return super.toString() + ":" + Dump.logoObject(value);
  }

  public LogoList report_1(final org.nlogo.nvm.Context context) {
    return value;
  }
}

File: netlogo-headless/src/main/prim/_constlist.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.Dump
import org.nlogo.core.{ LogoList, Pure }
import org.nlogo.nvm.{ Context, Reporter }

class _constlist(value: LogoList) extends Reporter with Pure {
  override def toString =
    super.toString + ":" + Dump.logoObject(value)
  override def report(context: Context): LogoList =
    report_1(context)
  def report_1(context: Context): LogoList =
    value
}
